lib/getNextJobs.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+/**
+ * Calculate the next jobs to execute, given a workflow and a trigger job
+ * @method getNextJobs
+ * @param  {Object}    workflow         Directed graph representation of workflow
+ * @param  {Object}    obj
+ * @param  {String}    obj.trigger      The triggering event (~pr, ~commit, jobName)
+ * @param  {String}    [obj.prNum]      The PR number (required when ~pr trigger)
+ * @return {Array}                      List of job names
+ */
+const getNextJobs = (workflow, obj) => {
+    const jobs = new Set();
+
+    if (!obj || !obj.trigger) {
+        throw new Error('Must provide a trigger');
+    }
+
+    if (obj.trigger === '~pr' && !obj.prNum) {
+        throw new Error('Must provide a PR number with "~pr" trigger');
+    }
+
+    workflow.edges.forEach((edge) => {
+        if (edge.src === obj.trigger) {
+            // Make PR jobs PR-$num:$cloneJob (not sure how to better handle multiple PR jobs)
+            jobs.add(obj.trigger === '~pr' ? `PR-${obj.prNum}:${edge.dest}` : edge.dest);
```

Comment:
I think we're going with `PR-${obj.prNum}-${edge.dest}`

Reply:
I think we should use `:` cuz jobName can actually contains/start with `-`.